agent/whatweb_agent.py

```diff
@@ -230,7 +230,11 @@ def _prepare_ip_targets(self, message: msg.Message) -> List[IPTarget]:
         if mask is None:
             network = ipaddress.ip_network(f"{host}")
         else:
-            version = message.data.get("version")
+            try:
+                ip = ipaddress.ip_address(host)
+                version = ip.version
+            except ValueError:
+                raise ValueError(f"Invalid IP address: {host}")
```

Comment:
I think you need to log it not re raise it

Reply:
By logging and returning an empty list, for example, we may inadvertently suppress an important error, leading to silent failures or unexpected behavior in the future.